code/tests/Hacl.Test.HMAC_DRBG.fst

```diff
@@ -0,0 +1,134 @@
+module Hacl.Test.HMAC_DRBG
+
+open FStar.HyperStack.ST
+open Test.Lowstarize
+
+open Lib.IntTypes
+
+open Hacl.HMAC_DRBG
+open Spec.HMAC_DRBG.Test.Vectors
+
+module D = Spec.Hash.Definitions
+module L = Test.Lowstarize
+module B = LowStar.Buffer
+
+#set-options "--max_fuel 0 --max_ifuel 0 --z3rlimit 100"
+
+(* FStar.Reflection only supports up to 8-tuples *)
+noextract
+let vectors_tmp = List.Tot.map 
+  (fun x -> x.a, h x.entropy_input, h x.nonce, h x.personalization_string, 
+         h x.entropy_input_reseed, h x.additional_input_reseed,
+         (h x.additional_input_1, h x.additional_input_2),
+         h x.returned_bits)
+  test_vectors
+
+%splice[vectors_low] (lowstarize_toplevel "vectors_tmp" "vectors_low")
+
+// Cheap alternative to friend Lib.IntTypes needed because Test.Lowstarize uses UInt8.t
+assume val declassify_uint8: squash (uint8 == UInt8.t)
+
+let vec8 = L.lbuffer UInt8.t
+
+let vector = D.hash_alg & vec8 & vec8 & vec8 & vec8 & vec8 & (vec8 & vec8) & vec8
+
+// This could replace TestLib.compare_and_print
+val compare_and_print: b1:B.buffer UInt8.t -> b2:B.buffer UInt8.t -> len:UInt32.t 
+  -> Stack bool
+  (requires fun h0 -> B.live h0 b1 /\ B.live h0 b2 /\ B.length b1 == v len /\ B.length b2 == v len)
+  (ensures  fun h0 _ h1 -> B.modifies B.loc_none h0 h1)
+let compare_and_print b1 b2 len =
+  push_frame();
+  LowStar.Printf.(printf "Expected: %xuy\n" len b1 done);
```

Comment:
Would be nice to have a printer using `PRIx8`.

Reply:
by design LowStar.Printf reduces before extraction to a series of linear calls to a series of specialized printers -- since we offer strictly more than what printf offers, it'd be hard to retain a call to lowstar.printf as a call to a C printf

Reply:
I meant that it would be nice to extend `LowStar.Printf` to add a new format string for hexadecimal `uint8` together with printers for scalars and buffers. I made a quick attempt, but ended up with an assertion failure only showing up in batch mode and didn't have time to debug it.

Reply:
Sorry I misunderstood. I think printers for scalars are decimal by default and printers for buffers are hex by default. Not sure if that helps in your case.

Reply:
No, we use decimal for both: https://github.com/FStarLang/kremlin/blob/master/kremlib/c/lowstar_printf.c#L18
Switching to hex for buffers would be a simple fix.

Reply:
I would be fine with that fix